### ISSUE TITLE
License? Rewrite?

### ISSUE THREAD

**Issue by ShadowMarker789** [Ormael13/CoCX#810]
Hi, 

I'm attempting to perform the impossible and rewrite Corruption of Champions in something that isn't Adobe™®© Flash™®© ActionScript™®© and was looking around for various mods/extensions to potentially include in the rewrite. 

Do you want your additions included? If not, I'll leave it by. So far, I've taken a look at Kitteh6660's codebase which looks mostly abandoned. :( 

That said, it doesn't look like any license was included on the original dump of Corruptions of Champions onto Github or any of the other git repos. 

What are you licensing your content as? Do we have a license for this? 

But, since I'm wanting to do a rewrite, I was thinking of making major overhauling changes to core mechanics (such as how combat, stats, works) ... do you have any suggestions or comments or things you'd like to see if things were to be rewritten from the ground-up? 

A discussion would have been better for this than raising an issue, but I couldn't see a discussions board for this.

**Comment by Clinkr1000** [comment]
It seems the original developers of this version of CoC have abandoned the project, and it is high time CoC got a reboot. As it is now the combat balance is atrocious, even in hardest difficulty.
I am willing to help convert it to HTML5, and remove some of that bloat and optimize things while at it.

**Comment by ShadowMarker789** [comment]
> It seems the original developers of this version of CoC have abandoned the project, and it is high time CoC got a reboot. As it is now the combat balance is atrocious, even in hardest difficulty. I am willing to help convert it to HTML5, and remove some of that bloat and optimize things while at it.

Right, but looking into it, when Fenoxo originally dumped the source, it had no license attached to it. 

This implies that everything in there is still under "All Rights Reserved" copyright and thus any sorts of reboots or ports would be afoul of copyright law. 

It is difficult for me to feel confident investing time into this when there's the potential of Fenoxo looking unkindly upon our efforts with disapproval and squashing it - as would be her right to do so.

**Comment by Clinkr1000** [comment]
i suppose that is true.

**Comment by ShadowMarker789** [comment]
> i suppose that is true.

That said, I'm still of the opinion that it would be a shame to see CoC lost to the sands of time in flash where no-one can run it. 

I am considering trying to port CoC over to React or some other sort of client-side application - javascript is similar-enough to AS3 that a lot of code can be ported quickly I think.

**Comment by nubesurrealista** [comment]
Btw this project is not abandoned, although it seems that it is now closed source, on the discord server there are always new builds.

**Comment by lavilao** [comment]
it seems pretty active on the master-wip branch

**Comment by Jtecx** [comment]
> It seems the original developers of this version of CoC have abandoned the project, and it is high time CoC got a reboot. As it is now the combat balance is atrocious, even in hardest difficulty.


Sorry for the late reply, but no. We haven't abandoned the project just yet. We're still highly active on the master-wip branch, and it's all open source. Near daily updates and patches on the discord as well are released, as well as autobuilds from github actions.

Regarding the original question however, they'd best ping Ormael on this directly. However, what I will say is that we have internally made some attempts to port the code to Typescript as well, though progress has stalled. 

Another developer on this project as well has built a custom engine, porting most of the stuff we have added and dropping some of the original content to free ourselves from the Fenco shackles but this has also stalled for the time being.

Again, apologies for the late replies. We don't look at github issues nearly as often as we should, with the majority of bug reports being done on discord instead.

@ShadowMarker789 since you're the original thread OP, pinging you on this as well, though you probably got the notice anyways.

**Comment by Clinkr1000** [comment]
where can i find the discord link?

**Comment by Jtecx** [comment]
> where can i find the discord link?

https://discord.gg/ZzmMjjT
